### ISSUE TITLE
OperationAnalysisContext should expose a SemanticModel

### ISSUE THREAD

**Issue by evildour** [dotnet/roslyn#23645]
**Version Used**: 2.6.0-beta3

**Expected Behavior**:
The `OperationAnalysisContext` for analyzers working with operations should expose a `SemanticModel`.

**Actual Behavior**:
`OperationAnalysisContext` does not expose a `SemanticModel`.

**Comment by jcouv** [comment]
@evildour Can you elaborate on why that would help?
Is there something you can't do at the moment? If this would help code become simpler, can you share a before/after example?

Tagging @heejaechang @mavasani to comment.

**Comment by evildour** [comment]
I had originally thought I needed it for an analyzer (see [here](https://github.com/dotnet/roslyn-analyzers/pull/1142/commits/49588100c1d6055df56727ae8d55eaadcb83ebdd#diff-b6cdf131aad4fc1b6f0c32466b1c48d2R33)), so I obtained it like so:
```csharp
var semanticModel = context.Compilation.GetSemanticModel(context.Operation.Syntax.SyntaxTree);
```
@mavasani asked me to file an issue here to have it exposed. Turns out I didn't need it anyway, but I still think it may be useful to expose in cases where an analyzer needs to drop down to the underlying syntax of an operation if it does not expose necessary information.

**Comment by mavasani** [comment]
Yes, operations are tied to syntax, which have a corresponding tree and SemanticModel. We have the semantic model in hand when creating an operation analysis context, so should expose it. We also expose it on other analysis contexts tied to syntax/tree.

**Comment by jcouv** [comment]
@evildour Thanks for the clarifications.
@mavasani I'll let you triage and assign. Cheers

**Comment by mavasani** [comment]
This will no longer be needed after https://github.com/dotnet/roslyn/pull/27922
